common/src/main/java/com/redhat/cloud/notifications/models/mappers/v1/endpoint/EndpointMapper.java

```diff
@@ -0,0 +1,136 @@
+package com.redhat.cloud.notifications.models.mappers.v1.endpoint;
+
+import com.redhat.cloud.notifications.models.CamelProperties;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointProperties;
+import com.redhat.cloud.notifications.models.SystemSubscriptionProperties;
+import com.redhat.cloud.notifications.models.WebhookProperties;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.EndpointDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.CamelPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.EndpointPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.SystemSubscriptionPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.WebhookPropertiesDTO;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingConstants;
+import org.mapstruct.Named;
+
+@Mapper(componentModel = MappingConstants.ComponentModel.CDI)
+public interface EndpointMapper {
+    /**
+     * Maps an internal endpoint's entity into a DTO.
+     * @param endpoint the internal entity to map.
+     * @return the mapped DTO.
+     */
+    @Mapping(source = "properties", target = "properties", qualifiedByName = "mapEntityProperties")
+    EndpointDTO toDTO(Endpoint endpoint);
+
+    /**
+     * Maps an endpoint's DTO to an internal entity.
+     * @param endpoint the DTO to map.
+     * @return the mapped internal entity.
+     */
+    @Mapping(source = "properties", target = "properties", qualifiedByName = "mapDTOProperties")
+    @Mapping(target = "accountId", ignore = true)
+    @Mapping(target = "behaviorGroupActions", ignore = true)
+    @Mapping(target = "notificationHistories", ignore = true)
+    @Mapping(target = "orgId", ignore = true)
+    @Mapping(target = "updated", ignore = true)
+    @Mapping(target = "serverErrorsSince", ignore = true)
+    Endpoint toEntity(EndpointDTO endpoint);
+
+    /**
+     * Maps a camel properties' internal entity to a DTO.
+     * @param camelProperties the internal entity to map.
+     * @return the mapped DTO.
+     */
+    CamelPropertiesDTO camelToDTO(CamelProperties camelProperties);
+
+    /**
+     * Maps a camel properties' DTO into an internal entity.
+     * @param camelPropertiesDTO the DTO to map.
+     * @return the mapped internal entity.
+     */
+    @Mapping(target = "basicAuthenticationSourcesId", ignore = true)
+    @Mapping(target = "bearerAuthentication", ignore = true)
```

Comment:
Is that needed because of `getBearerAuthentication` and `setBearerAuthentication` in `CamelProperties`?

Reply:
Exactly, yes. My idea was that once we have the DTOs in place we will be able to refactor the internal model for the secrets to make it simpler and not force every property to have all these methods implemented...